extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java

```diff
@@ -299,4 +328,39 @@ private String stripSlashes(String uri) {
         return result;
     }
 
+    private class ForwardedValues {
+        String forwardedScheme = scheme == null ? "" : scheme;
+        String forwardedHost = host == null ? "" : host;
+        int forwardedPort = port;
+        String forwardedUri = uri == null ? "" : uri;
+        String forwardedRemoteHost = "";
+        int forwardedRemotePort = -1;
+    }
+
+    private class XForwardedValues {
+        String xForwardedScheme = scheme == null ? "" : scheme;
+        String xForwardedHost = host == null ? "" : host;
+        int xForwardedPort = port;
+        String xForwardedUri = uri == null ? "" : uri;
+        String xForwardedRemoteHost = "";
+        int xForwardedRemotePort = -1;
+    }
```

Comment:
Wouldn't it be a lot simpler using one record for both and using equals on the records?

Reply:
@gsmet Some or all of there properties are set in the `allowForwarded` and `allowXForwarded` cases, I found it was easier this way to keep corresponding properties, the code for handling both cases is not quite symmetrical, For `allowxForwaded` I started with declaring directly, without any bean wrappers:

```
String xForwardedScheme = scheme;
        String xForwardedHost = host;
        int xForwardedPort = port;
        String xForwardedUri = uri;
        String xForwardedRemoteHost = null;
        int xForwardedRemotePort = -1;
```
and the same for the `allowForwarded` case, and got confused quickly as it is not clear when a given property can be null, and then when both code branches are finished, they have to be compared, so this is why I ended doing it this way.

I'm all for trying to make it neater, I'm just not sure if, without records, it would be much simpler to have a single class and a manual `equals` implementation... but yeah, I'm happy to tune it further...


What do you think ?

Reply:
Ah, your problem is the immutable nature of the record and the fact you want to initialize them property per property?

Not a big fan but I suppose having a builder + creating a record is not ideal in a runtime path that could end up being hot.

In any case, I'm not sure I would go with two different containers given they are identical. You could have one container with a proper equals and use that.

But yeah, do as you think is best as it's important this code doesn't break.

Reply:
@gsmet Yeah, I can collapse them into a single one at least

Reply:
@gsmet It should look better now, thanks